[APP-4064] Account Box Cards Draft

angelojcruzjr opened the pull request:
### Description
> Please provide a brief description of your changes below.

Created two new components, one is an `atom` called `CardContainer` and the other is a `molecule` titled `CardContainerWithRows`.

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

Run storybook and checkout the components! Also, more importantly, peep the code for how I set these up 👀.

### Comments
> Any additional comments that will aid with the review process.

Very open to changing the names of these, just put in whatever popped into my head  😅

**Note:**
- No tests have been written yet
- No component interactivity yet

Want to make sure these are on the right track before I continue! Original comps can be found in the Jira ticket.

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/APP-4064)

RohitRajendran commented on components/atoms/CardContainer/CardContainer.js:
```diff
@@ -0,0 +1,57 @@
+  render() {
+    const {title, cardContent, isUiAccount} = this.props;
+
+    let accountClass = 'uic--card-container-title ';
```
Check out the classnames package that we use in other components like the FormSummary. It makes it much cleaner to do conditional classes

angelojcruzjr replied:
No longer an issue since we don't ask for isUiAccount anymore 👀

RohitRajendran commented on components/atoms/CardContainer/CardContainer.js:
```diff
@@ -0,0 +1,57 @@
+  /** The message to be displayed */
+  title: PropTypes.string.isRequired,
+  /** Is this a UI account? */
+  isUiAccount: PropTypes.bool,
```
Instead of a toggle to change the image, I think we need to let this be flexible for whatever image that wants to be used.

JamesIves replied:
I agree with @RohitRajendran  - I'm not a fan of naming this `isUiAccount` either, as this library is open sourced we should be using agnostic language here, and throughout the whole component.

angelojcruzjr replied:
Updated logic in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960 so that we just accept an image URL

RohitRajendran commented on components/atoms/CardContainer/CardContainer.js:
```diff
@@ -0,0 +1,57 @@
+import './CardContainer.scss';
+
+/** Renders a list which allows  */
+class CardContainer extends Component {
```
I think we should rename this so that it doesn't get confused with the cards for the McG flow. I'm not really sure what though, maybe InfoBox?

angelojcruzjr replied:
Renamed to InfoBox in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

RohitRajendran commented on components/molecules/CardContainerWithRows/CardContainerWithRows.js:
```diff
@@ -0,0 +1,120 @@
+/** @module Ranking */
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
```
```suggestion
import React, {Component, Fragment} from 'react';
```

So that you can just do `Fragment` instead of `React.Fragment`.

angelojcruzjr replied:
Updated to use just Fragment in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

RohitRajendran commented on components/molecules/CardContainerWithRows/CardContainerWithRows.js:
```diff
@@ -0,0 +1,120 @@
+  rowData: PropTypes.arrayOf(
+    PropTypes.shape({
+      /** Key for this piece of data */
+      key: PropTypes.string.isRequired,
```
We should make this optional and just use label as the key if none provided

angelojcruzjr replied:
Updated to be optional in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

RohitRajendran commented on components/molecules/CardContainerWithRows/CardContainerWithRows.js:
```diff
@@ -0,0 +1,120 @@
+      /** Label for this piece of data */
+      label: PropTypes.string.isRequired,
+      /** Value for this piece of data */
+      value: PropTypes.number.isRequired,
```
Value should be a string. We should get rid of the format object because the value should be the preformatted string

angelojcruzjr replied:
Updated to use string values that the user passes in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

RohitRajendran commented on components/molecules/CardContainerWithRows/CardContainerWithRows.js:
```diff
@@ -0,0 +1,120 @@
+    append: PropTypes.string,
+  }),
+  /** Optional override statement */
+  override: PropTypes.shape({
```
Lets call this the footer and allow JSX to be passed through

angelojcruzjr replied:
Renamed to footer in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/molecules/CardContainerWithRows/CardContainerWithRows.js:
```diff
@@ -0,0 +1,120 @@
+}
+
+CardContainerWithRows.propTypes = {
+  rowData: PropTypes.arrayOf(
```
Missing prop comment here

angelojcruzjr replied:
Added in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/molecules/CardContainerWithRows/CardContainerWithRows.js:
```diff
@@ -0,0 +1,120 @@
+      value: PropTypes.number.isRequired,
+    }).isRequired,
+  ).isRequired,
+  /** Optional formatting */
```
Can you make these comments a bit more descriptive across the board? Optional formatting for what?

angelojcruzjr replied:
Went in and re-did all the comments in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/molecules/CardContainerWithRows/CardContainerWithRows.scss:
```diff
@@ -0,0 +1,28 @@
+
+.uic--card-container-with-rows {
+  align-items: center;
+  display: flex;
```
There's a series of helper classes you can use for these, ie `uic--d-flex` etc.

RohitRajendran replied:
I still don't agree with using helper classes if you anyways need to use a custom class

JamesIves replied:
🥊

angelojcruzjr replied:
I used a helper class in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960 ...sorry @RohitRajendran 🙈

RohitRajendran replied:
How could you @angelojcruzjr!? 💔

JamesIves commented on components/atoms/CardContainer/CardContainer.story.js:
```diff
@@ -0,0 +1,34 @@
+  .add('default', () => (
+    <CardContainer
+      title={text('Text', 'My 401k Account')}
+      isUiAccount={boolean('IsUiAccount', false)}
```
Can you use the `defaultProps` function model for this?

angelojcruzjr replied:
Updated both components to use `defaultProps` function model in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/atoms/CardContainer/CardContainer.scss:
```diff
@@ -0,0 +1,47 @@
+    position: relative;
+
+    // so that the child node does not cause
+    // the parents node onMouseOut event to fire
```
Does this need to be a wildcard though? I'd prefer if we wrapped it in a class and assigned the property there. This isn't a pattern I'm fond of.

angelojcruzjr replied:
I just left this in the child class in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960. I think it makes more sense since we control what goes inside here so we don't necessarily need to account for all descending children.

JamesIves commented on components/atoms/CardContainer/CardContainer.js:
```diff
@@ -0,0 +1,57 @@
+};
+
+CardContainer.defaultProps = {
+  isUiAccount: true,
```
Same comment about the language. Should this be defaulted to true?

angelojcruzjr replied:
No longer applicable due to logic changes in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/atoms/CardContainer/CardContainer.js:
```diff
@@ -0,0 +1,57 @@
+    super(props);
+
+    this.state = {
+      title: props.title,
```
Why do we need to bind `props.title` to state here?  It looks like you can just rely on props and make this a functional component.

angelojcruzjr replied:
You're right I wasn't using state for anything in either component. Updated both to be pure components in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/atoms/CardContainer/CardContainer.scss:
```diff
@@ -0,0 +1,47 @@
+      background-image: url('../../../assets/icons/investment.svg');
+    }
+
+    &.ui-account {
```
Same comment regarding language.

angelojcruzjr replied:
No longer applicable due to logic changes in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/atoms/CardContainer/CardContainer.scss:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    &.non-ui-account {
+      background-image: url('../../../assets/icons/investment.svg');
```
These svgs should probably be passed into the component as opposed to hardcoding them. What if the user is using this in an application other than ui?

angelojcruzjr replied:
No longer applicable due to logic changes in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/molecules/CardContainerWithRows/CardContainerWithRows.js:
```diff
@@ -0,0 +1,120 @@
+import './CardContainerWithRows.scss';
+
+/**
+ * Utility function that formats our values
```
Confused about what this does by this comment. Which values?

angelojcruzjr replied:
Updated all comments in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

JamesIves commented on components/molecules/CardContainerWithRows/CardContainerWithRows.js:
```diff
@@ -0,0 +1,120 @@
+  return returnVal;
+}
+
+/** Renders a list which allows  */
```
Incomplete comment

angelojcruzjr replied:
Updated all comments in https://github.com/UnitedIncome/components/pull/867/commits/21c1882106753193b3b995f2e26d6b5578cca960

angelojcruzjr commented:
@RohitRajendran @JamesIves made a bunch of updates if you want to take another pass at this whenever y'all have free time. Going to work on tests in the meantime!

RohitRajendran commented on components/atoms/InfoBox/InfoBox.scss:
```diff
@@ -0,0 +1,34 @@
+    position: relative;
+
+    .uic--info-box-title-text {
+      @extend %uic--header-2-text;
```
This should just get applied as a class instead of extending it.

angelojcruzjr replied:
Done and done in https://github.com/UnitedIncome/components/pull/867/commits/f44defbd99fd470248924192d3a00b8791ce5ad6

RohitRajendran commented on components/atoms/InfoBox/InfoBox.story.js:
```diff
@@ -0,0 +1,53 @@
+  title: text('title', title),
+  imageURL: text('imageURL', imageURL),
+  infoBoxContent: object('infoBoxContent', infoBoxContent),
+  onChange: (name, value) => store.set({[name]: value}),
```
What do you need the onChange for?

angelojcruzjr replied:
I need to read over these storybook docs in detail 😞 - thought it was needed for component interactivity but turns out that's not the case. Got rid of it in https://github.com/UnitedIncome/components/pull/867/commits/f44defbd99fd470248924192d3a00b8791ce5ad6

RohitRajendran commented on components/molecules/InfoBoxFormattedRows/InfoBoxFormattedRows.js:
```diff
@@ -0,0 +1,48 @@
+import './InfoBoxFormattedRows.scss';
+
+/** Renders an InfoBox with the title formatted the same way the
+ *  basic title us, but with a well-defined row structure where data
```
`basic title us` typo

angelojcruzjr replied:
More better grammar in https://github.com/UnitedIncome/components/pull/867/commits/f44defbd99fd470248924192d3a00b8791ce5ad6

RohitRajendran commented on components/molecules/InfoBoxFormattedRows/InfoBoxFormattedRows.js:
```diff
@@ -0,0 +1,48 @@
+            <div className="uic--mcg-subhead-text">{row.value}</div>
+          </div>
+        ))}
+        {footer ? footer : ''}
```
If the user is passing in a footer, the divider line thing should get added. That shouldn't be something the user has to include

RohitRajendran replied:
We might want to actually make this functionality for the basic InfoBox

angelojcruzjr replied:
Added in the line and made the footer a part of the broader `InfoBox` component in https://github.com/UnitedIncome/components/pull/867/commits/f44defbd99fd470248924192d3a00b8791ce5ad6

angelojcruzjr commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,38 @@
+  /** The main title that will displayed near the top of our InfoBox */
+  title: PropTypes.string.isRequired,
+  /** The infoBoxContent JSX to be displayed below the title */
+  infoBoxContent: PropTypes.node.isRequired,
```
Changed to `content` in https://github.com/UnitedIncome/components/pull/867/commits/f44defbd99fd470248924192d3a00b8791ce5ad6

angelojcruzjr commented on components/atoms/InfoBox/InfoBox.scss:
```diff
@@ -0,0 +1,34 @@
+$background-size: 17rem;
+
+.uic--info-box {
+  @include antialias;
```
Ah I see - this makes sense. It was not rendering to spec based on the comps I was seeing but that's because we will antialias on the app itself and that should be good to go. Changed in https://github.com/UnitedIncome/components/pull/867/commits/f44defbd99fd470248924192d3a00b8791ce5ad6

angelojcruzjr commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,38 @@
+  }
+}
+
+InfoBox.propTypes = {
```
Working on this still! Watch for the next commit.

angelojcruzjr replied:
This is in as of https://github.com/UnitedIncome/components/pull/867/commits/13a29992ae9f227b4d94ca78a84cefb49ff437b0

JamesIves commented on components/molecules/InfoBoxFormattedRows/InfoBoxFormattedRows.mdx:
```diff
@@ -0,0 +1,82 @@
+
+## Example 🚀
+
+_Note:_ The imageURL should link to whatever custom image you would like to pass into the InfoBox.
```
This is super nitpicky, but can we just add this as a prop type comment? There's no real need to add a note here when it will show up in the table below.

angelojcruzjr replied:
Updated README in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on components/molecules/InfoBoxFormattedRows/InfoBoxFormattedRows.js:
```diff
@@ -0,0 +1,45 @@
+ *  structure where data labels are on the left and their associated
+ *  values on the right.
+ */
+class InfoBoxFormattedRows extends Component {
```
This could  be a functional component as you're using none of the reacts class component lifecycle methods besides render, ie `const InfoBoxFormattedRows = () => ( <div></div>)`

angelojcruzjr replied:
Updated to be functional in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on components/atoms/InfoBox/InfoBox.mdx:
```diff
@@ -0,0 +1,45 @@
+
+## Example 🚀
+
+_Note:_ The imageURL should link to whatever custom image you would like to pass into the InfoBox.
```
Same note about the readme

angelojcruzjr replied:
Updated readme in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,55 @@
+/** Renders a basic card with a title component on top and
+ *  description/more information below the title portion.
+ */
+class InfoBox extends Component {
```
Same comment about the other functional component

angelojcruzjr replied:
Made functional in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,55 @@
+            {title}
+          </div>
+        </div>
+        <div className="uic--info-box-information-area">
```
If this is the footer of the component, using the `<footer>` element may be more appropriate.

angelojcruzjr replied:
Yup better semantics. Changed in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,55 @@
+/** @module Ranking */
```
```suggestion
/** @module InfoBox */
```

angelojcruzjr replied:
Woops. Updated in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on components/atoms/InfoBox/InfoBox.scss:
```diff
@@ -0,0 +1,44 @@
+
+.uic--info-box {
+  border: 1px solid $light-gray;
+  cursor: default;
```
What is the need for `cursor: default` here?

angelojcruzjr replied:
Leftover from a copy and paste. Got rid of it in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,55 @@
+    };
+
+    return (
+      <div className="uic--info-box">
```
`<aside>` maybe? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/aside

angelojcruzjr replied:
So I looked into the docs and from what I could tell asides are supposed to be used for tangentially related information to the main page...are you saying that this entire InfoBox concept would be an aside? I can see an argument for this. Just want to confirm before I change. Would like to hear @RohitRajendran 's opinion as well 😇

JamesIves replied:
Yeah, as it contains content which is related to the page but not the sole purpose of the page I feel it may make sense in this scenario. `<section />` may also be applicable for these. 

I'm not attached to either idea, just a possible area of improvement.

angelojcruzjr replied:
Made it `<aside />` in https://github.com/UnitedIncome/components/pull/867/commits/13a29992ae9f227b4d94ca78a84cefb49ff437b0

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,55 @@
+          className="uic--info-box-title-area"
+          style={imageURL ? cardStyle : ''}
+        >
+          <div className="uic--info-box-title-text uic--header-2-text">
```
I feel like this should be some kind of header tag instead of a div as it defines content.

angelojcruzjr replied:
Changed to header in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on assets/icons/ui-account.svg:
```diff
@@ -0,0 +1,13 @@
+<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" width="265" height="160" viewBox="0 0 265 160">
```
If this isn't actually used for anything let's remove it and place a hardcoded link to an image for the previews.

angelojcruzjr replied:
Removed and replaced with a hardcoded svg link in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

JamesIves commented on components/atoms/InfoBox/InfoBox.scss:
```diff
@@ -0,0 +1,44 @@
+      left: 2rem;
+      pointer-events: none;
+      position: absolute;
+      right: 6rem;
```
I can't seem to figure out what this does

angelojcruzjr replied:
So this positions the `title` to the position it is currently in

JamesIves commented on components/atoms/InfoBox/InfoBox.scss:
```diff
@@ -0,0 +1,44 @@
+      pointer-events: none;
+      position: absolute;
+      right: 6rem;
+      text-align: left;
```
Probably don't need to define the text alignment here

angelojcruzjr replied:
Correct - extraneous here. Got rid of it in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

RohitRajendran commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,55 @@
+        <div className="uic--info-box-information-area">
+          {content}
+          {footer ? (
+            <Fragment>
```
You can just do `footer && (` instead of the ternary

angelojcruzjr replied:
🤦🏽‍♂️true. Updated in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

RohitRajendran commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,55 @@
+          {content}
+          {footer ? (
+            <Fragment>
+              <div className="uic--info-box-formatted-row border-top" />
```
Why not an hr?

angelojcruzjr replied:
Updated in https://github.com/UnitedIncome/components/pull/867/commits/0e081efa568fc65b91be2baff1a8ba08819e8f5f

angelojcruzjr commented:
@JamesIves @RohitRajendran made another round of updates that should address all the feedback up to this point 😇

RohitRajendran commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,59 @@
+  });
+
+  return (
+    <aside className={infoBoxClass}>
```
Also allow passing in style

angelojcruzjr replied:
Allowed it here and in `InfoBoxFormattedRows` with changes in https://github.com/UnitedIncome/components/pull/867/commits/7697e6518948ab700c97b7d5bb7c6d6e4b6fdd90

RohitRajendran commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,59 @@
+        className="uic--info-box-title-area"
+        style={imageURL ? cardStyle : ''}
+      >
+        <h1 className="uic--info-box-title-text uic--header-2-text">{title}</h1>
```
I don't think h1 is right for this. There should only be one h1 on the page.

angelojcruzjr replied:
Hmmm. Which would you say is appropriate here? `h2`?

angelojcruzjr replied:
Made `h2` in https://github.com/UnitedIncome/components/pull/867/commits/529165bfc44344c47c80e8fc57ee90424707cc07

RohitRajendran commented on components/atoms/InfoBox/InfoBox.mdx:
````diff
@@ -0,0 +1,60 @@
+the card content and the footer. Below is an example footer that would create a nicely formatted row:
+
+```javascript
+<div className="uic--info-box-formatted-row uic--d-flex uic--justify-content-between">
````
I think this is one of the situations that it's a lot better to have those flex properties be part of that custom class so that users don't have to add a bunch of classes to get the same styling

@JamesIves

angelojcruzjr replied:
I agree - let's see if we can get consensus 😊

angelojcruzjr replied:
Made this change in https://github.com/UnitedIncome/components/pull/867/commits/7697e6518948ab700c97b7d5bb7c6d6e4b6fdd90

JamesIves replied:
I agree in this case

RohitRajendran commented on components/atoms/InfoBox/InfoBox.scss:
```diff
@@ -0,0 +1,42 @@
+$background-size: 17rem;
+$container-min-height: 2.4rem;
+
+hr {
```
hr styling should be scoped inside of uic--info-box so that the styling doesn't bleed

angelojcruzjr replied:
Updated in https://github.com/UnitedIncome/components/pull/867/commits/7697e6518948ab700c97b7d5bb7c6d6e4b6fdd90

RohitRajendran commented on components/molecules/InfoBoxFormattedRows/InfoBoxFormattedRows.js:
```diff
@@ -0,0 +1,54 @@
+  );
+};
+
+InfoBoxFormattedRows.propTypes = {
```
Need to also allow footer to be passed through

RohitRajendran replied:
And style

angelojcruzjr replied:
> Need to also allow footer to be passed through

So I currently have the `footer` be passed through the `InfoBox` shell component. Are you saying it should go in `InfoBoxFormattedRows`?

RohitRajendran replied:
Ohhh jk, I was thinking of this wrong

RohitRajendran commented on components/molecules/InfoBoxFormattedRows/InfoBoxFormattedRows.js:
```diff
@@ -0,0 +1,54 @@
+import classNames from 'classnames';
+
+/**
+ * Renders an InfoBox with the title formatted the same way the
```
This needs to be updated

angelojcruzjr replied:
Updated in https://github.com/UnitedIncome/components/pull/867/commits/529165bfc44344c47c80e8fc57ee90424707cc07

RohitRajendran commented on components/atoms/InfoBox/InfoBox.mdx:
```diff
@@ -0,0 +1,60 @@
+InfoBox comes with several helper styles that make it easy for you to adhere to
+the style guide - especially when adding a footer. If you want to add a nicely aligned
+row such as what is seen in `InfoBoxFormattedRows`, add the following classes to the row parent element:
+`uic--info-box-formatted-row uic--d-flex uic--justify-content-between`. Labels can then be wrapped
```
I think this was actually more necessary/useful when InfoBoxFormattedRows was more of a wrapper component but since it's meant to be plugged into the content/footer. I don't think this is really necessary to call out anymore. Maybe we should change this section to talk about how you can use the InfoBoxFormattedRows with this

angelojcruzjr replied:
Updated in https://github.com/UnitedIncome/components/pull/867/commits/7697e6518948ab700c97b7d5bb7c6d6e4b6fdd90

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,61 @@
+ */
+const InfoBox = (props) => {
+  const {title, content, imageURL, footer, className, style} = props;
+  let appliedStyle = {};
```
This is minor (and optional), but this could probably be cleaned up a little bit.

```
const appliedStyles = {...style}

if (imageURL) {
   appliedStyles.backgroundImage: `url(${imageURL})`
}
```

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,61 @@
+
+  const infoBoxClass = classNames({
+    'uic--info-box': true,
+    className,
```
I think you need to pass `className` in as the second argument here.

Example:

```
    const containerClasses = classNames(
      {
        'uic--itemization-box': true,
      },
      className,
    );
```

JamesIves commented on components/atoms/InfoBox/InfoBox.mdx:
```diff
@@ -0,0 +1,52 @@
+
+## Props 🔧
+
+<Props of={InfoBox} />
```
In some of the components we have a `Related` header that links to similar things. Can we add that here? 

Here's the template:

```suggestion
<Props of={InfoBox} />

## Related ✨

- [Checkboxes](/?path=/story/molecules-checkboxes--default)
```

We should link to InfoBoxFormatted and do the same in the other readme.

RohitRajendran commented on components/molecules/InfoBoxFormattedRows/InfoBoxFormattedRows.mdx:
```diff
@@ -0,0 +1,78 @@
+import {Preview, Props} from '@storybook/addon-docs/blocks';
```
I don't think this component should be a molecule anymore. Seems more like an atom

angelojcruzjr replied:
😱I'll move it over before merging.

JamesIves replied:
I'm looking forward to re-organizing Storybook, most of the library definitions are way off 😞

angelojcruzjr commented:
@RohitRajendran made changes that allow for the InfoBox to have the same functionality as what is currently in the app. Also moved it over to Atoms and added tests/a story for a small utility function that I created. I'll merge this once you give your blessing 🙏🏽

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,161 @@
+import classNames from 'classnames';
+import {formatEllipsis} from '../../utilities/FormatUtils/FormatUtils';
+
+const CHAR_COUNT = 25;
```
Can we make this a prop and just set this as the default value?

angelojcruzjr replied:
Yeah that's a lot better. Went ahead and made this an optional prop in https://github.com/UnitedIncome/components/pull/867/commits/49076394869a5a2bd9eedb386d56ec73139ae885

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,161 @@
+  /** Additional style properties to apply to the container. */
+  style: PropTypes.objectOf(PropTypes.string),
+  /** Optional onClick handler. */
+  onClick: PropTypes.func,
```
This doesn't appear to be optional;

```
  handleClick(event) {
    event.stopPropagation();

    return this.props.onClick();
  }
```

If `onClick` isn't provided it's going to error. We should check its existence before firing it. Either that or mark thisprop has required.

angelojcruzjr replied:
Updated to check for existence before firing in latest push (https://github.com/UnitedIncome/components/pull/867/commits/49076394869a5a2bd9eedb386d56ec73139ae885)

JamesIves commented on components/utilities/FormatUtils/FormatUtils.mdx:
````diff
@@ -41,3 +41,15 @@ FormatUtils.formatCurrencyNoDecimal(2000);
+```javascript
+import {FormatUtils} from '@unitedincome/components';
+
+FormatUtils.formatEllipsis('This is a very, very long string', 25);
````
Is it ok for the title to wrap on small screen sizes?

<img width="380" alt="Screen Shot 2019-11-25 at 12 51 03 PM" src="https://user-images.githubusercontent.com/10888441/69565124-4cfc7500-0f82-11ea-927f-2f035f14da7a.png">

I typically like to use CSS for things like this because it allows you specify a specific width percentage that should display prior to the ellipsis generation, as opposed to a hard character limit. Both methods have their drawbacks though.

angelojcruzjr replied:
So I went back and forth on how this should look and after talking w/ Monique @RohitRajendran and I decided to just use the functionality that is currently in place of the app which is using a hard character limit. I have a short video of the current functionality that I can send you if you're curious (github won't let me upload .mov's 😞) but I can confirm this is how it currently looks. This will probably change in the future but for now we are just keeping it status quo, so to speak.

JamesIves replied:
Sounds good to me

JamesIves commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,161 @@
+};
+
+InfoBox.defaultProps = {
+  selectable: false,
```
Does this need to be a default prop? Isn't `selectable` just going to default to falsey?

angelojcruzjr replied:
That's true. Got rid of it in https://github.com/UnitedIncome/components/pull/867/commits/49076394869a5a2bd9eedb386d56ec73139ae885

RohitRajendran commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,164 @@
+          onMouseEnter={this.handleMouseOver}
+          onMouseLeave={this.handleMouseOut}
+        >
+          <h2 className="uic--info-box-title-text uic--header-2-text">
```
If you look at the comps, I don't think h2 makes sense from an information hierarchy standpoint.

RohitRajendran replied:
Since it can depend on the page that it's one, I think we should skip using h tags for this.

RohitRajendran commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,164 @@
+  /** Optional footer JSX to be displayed below the content. */
+  footer: PropTypes.node,
+  /** Optionally make the InfoBox card appear clickable/selectable. */
+  selectable: PropTypes.bool,
```
Can't you just infer this from if onClick was passed through?

RohitRajendran commented on components/atoms/InfoBox/InfoBox.js:
```diff
@@ -0,0 +1,164 @@
+    }
+
+    if (onClick) {
+      containerProps.onClick = this.handleClick;
```
Can't you just pass `this.props.onClick` directly instead of using a wrapper function?

JamesIves replied:
@angelojcruzjr Can we address these in a separate PR?

angelojcruzjr replied:
Si. Will create a new one for @RohitRajendran 's comments
